aws/resource_aws_guardduty_publishing_destination.go

```diff
@@ -0,0 +1,192 @@
+		}
+		resp, err := conn.DescribePublishingDestination(input)
+		if err != nil {
+			return nil, "failed", err
+		}
+		return resp, *resp.Status, nil
+	}
+}
+
+func resourceAwsGuardDutyPublishingDestinationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	destination_id, detector_id, err_state_read := decodeGuardDutyPublishDestinationID(d.Id())
+
+	if err_state_read != nil {
+		return err_state_read
+	}
+
+	input := &guardduty.DescribePublishingDestinationInput{
+		DetectorId:    aws.String(detector_id),
+		DestinationId: aws.String(destination_id),
+	}
+
+	log.Printf("[DEBUG] Reading GuardDuty publishing destination: %s", input)
+	gdo, err := conn.DescribePublishingDestination(input)
+	if err != nil {
+		if isAWSErr(err, guardduty.ErrCodeBadRequestException, "The request is rejected because the input detectorId is not owned by the current account.") {
+			log.Printf("[WARN] GuardDuty publishing destination: %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Reading GuardDuty publishing destination: '%s' failed: %s", d.Id(), err.Error())
+	}
+
+	d.Set("detector_id", detector_id)
+	d.Set("destination_type", gdo.DestinationType)
+	d.Set("kms_key_arn", gdo.DestinationProperties.KmsKeyArn)
+	d.Set("destination_arn", gdo.DestinationProperties.DestinationArn)
+	d.Set("status", gdo.Status)
+	return nil
+}
+
+func resourceAwsGuardDutyPublishingDestinationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	destination_id, detector_id, err_state_read := decodeGuardDutyPublishDestinationID(d.Id())
+
+	if err_state_read != nil {
+		return err_state_read
+	}
+
+	input := guardduty.UpdatePublishingDestinationInput{
+		DestinationId: aws.String(destination_id),
+		DetectorId:    aws.String(detector_id),
+		DestinationProperties: &guardduty.DestinationProperties{
+			DestinationArn: aws.String(d.Get("destination_arn").(string)),
+			KmsKeyArn:      aws.String(d.Get("kms_key_arn").(string)),
+		},
+	}
+
+	log.Printf("[DEBUG] Update GuardDuty publishing destination: %s", input)
+	_, err := conn.UpdatePublishingDestination(&input)
+	if err != nil {
+		return fmt.Errorf("Updating GuardDuty publishing destination '%s' failed: %s", d.Id(), err.Error())
+	}
+
+	return resourceAwsGuardDutyPublishingDestinationRead(d, meta)
+}
+
+func resourceAwsGuardDutyPublishingDestinationDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	destination_id, detector_id, err_state_read := decodeGuardDutyPublishDestinationID(d.Id())
+
+	if err_state_read != nil {
+		return err_state_read
+	}
+
+	input := guardduty.DeletePublishingDestinationInput{
+		DestinationId: aws.String(destination_id),
+		DetectorId:    aws.String(detector_id),
+	}
+
+	log.Printf("[DEBUG] Delete GuardDuty publishing destination: %s", input)
+	_, err := conn.DeletePublishingDestination(&input)
```

Comment:
What happens if the PublishingDestination has been deleted outside Terraform?
Do we need the `if isAWSErr(err, guardduty.ErrCodeBadRequestException,...` check here?